app/controllers/volunteers_controller.rb

```diff
@@ -129,14 +121,11 @@ def create
     return unless check_permissions(@volunteer)
     # can't set admin bits from CRUD controls
     @volunteer.admin = false
-    @volunteer.assignments.each{ |assingment| assingment.admin = false }
+    @volunteer.assignments.each { |assignment| assignment.admin = false }
+
     if @volunteer.save
       flash[:notice] = 'Created successfully.'
-      unless session[:my_return_to].nil?
-        redirect_to(session[:my_return_to])
-      else
-        index
-      end
+      redirect_to(session[:my_return_to] || index)
     else
       flash[:error] = "Didn't save successfully :(. #{@volunteer.errors.full_messages.to_sentence}"
```

Comment:
as per our conversation about explicit returning after redirects do you need one right here? won't it try to render after its redirected?